src/fallback.rs

```diff
@@ -840,22 +840,22 @@ impl fmt::Debug for Literal {
 fn token_stream(mut input: Cursor) -> PResult<TokenStream> {
     let mut trees = Vec::new();
     loop {
-        let input_no_ws = skip_whitespace(input);
-        if input_no_ws.rest.len() == 0 {
-            break;
-        }
-        if let Ok((a, tokens)) = doc_comment(input_no_ws) {
+        if let Ok((a, tokens)) = doc_comment(input) {
             input = a;
             trees.extend(tokens);
             continue;
         }
 
-        let (a, tt) = match token_tree(input_no_ws) {
+        let (a, tt) = match token_tree(input) {
```

Comment:
Help me understand this change: This causes all whitespace and comments to be parsed twice as many times, doesn't it? The previous behavior made 3 calls to skip_whitespace but only 1 of them would do any work. The new behavior makes 2 calls to skip_whitespace where the second one redoes all the work already done by the first one. Seems like which one is better would depend heavily on the characteristics of the input.

Instead we might want to split some of these functions like what we have for `symbol_leading_ws` vs `symbol` below, and always call the right one depending on whether whitespace has already been processed at the call site.

Reply:
Yes, it should go where it was. 
I want to remove the call to `skip_whitespace` at `spanned`. I just removed it to make sure it worked since the logic is complex.